hooks/subnamespace.go

```diff
@@ -67,15 +70,44 @@ func (v *subNamespaceValidator) Handle(ctx context.Context, req admission.Reques
 		return admission.Errored(http.StatusInternalServerError, err)
 	}
 
-	if ns.Labels[constants.LabelType] == constants.NSTypeRoot || ns.Labels[constants.LabelParent] != "" {
-		return admission.Allowed("")
+	if ns.Labels[constants.LabelType] != constants.NSTypeRoot && ns.Labels[constants.LabelParent] == "" {
+		return admission.Denied(fmt.Sprintf("namespace %s is neither a root nor a sub namespace", ns.Name))
+	}
+
+	ok, err := v.notMatchingNamingPolicy(ctx, sn)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+	if !ok {
+		return admission.Denied(fmt.Sprintf("namespace %s is not match naming policies", ns.Name))
+	}
+
+	return admission.Allowed("")
+}
+
+func (v *subNamespaceValidator) notMatchingNamingPolicy(ctx context.Context, sn *accuratev1.SubNamespace) (bool, error) {
```

Comment:
The regular expression should be pre-compiled when loaded from the configuration file,
so that no error would be returned during execution.